### PR TITLE
Enable Saving Fairs

### PR DESCRIPTION
**Pull request by ashleyjelks**
This PR supports a mutation enabling users to save fairs to their user profile. 

[Links to LD-102](https://artsyproduct.atlassian.net/browse/LD-102)

kapture 2019-02-06 at 14 28 14

#skip_new_tests

### CODE REVIEW

File: src/lib/Scenes/Fair/Components/FairHeader/index.tsx [hunk 1]
```diff
@@ -16,6 +17,11 @@ interface Props {
   onSaveShowPressed?: () => Promise<void>
   viewAllExhibitors: () => void
   viewAllArtists: () => void
+  relay: RelayProp
+}
+
+interface State {
+  isSavedFairStateUpdating: boolean
```
Comment (damassi) [inline src/lib/Scenes/Fair/Components/FairHeader/index.tsx]: 👍

File: src/lib/Scenes/Fair/Components/FairHeader/index.tsx [hunk 2]
```diff
@@ -112,9 +176,15 @@ export class FairHeader extends React.Component<Props> {
         </BackgroundImage>
         <Spacer mt={2} />
         <Box mx={2}>{this.getContextualDetails()}</Box>
-        <Box px={2}>
+        <Box px={2} width={375} height={95}>
```
Comment (kierangillen) [inline src/lib/Scenes/Fair/Components/FairHeader/index.tsx]: Hey @ashleyjelks you have a set width here. This Box should be 100% width. I moved the one I added into a styled component to add `width: 100%;`

Comment (l2succes) [reply src/lib/Scenes/Fair/Components/FairHeader/index.tsx]: Actually I wouldn't hardcode a height here either. The button component should know it's own size

File: src/lib/Scenes/Fair/Components/FairHeader/index.tsx [hunk 3]
```diff
@@ -85,11 +93,67 @@ export class FairHeader extends React.Component<Props> {
     Switchboard.presentNavigationViewController(this, url)
   }
 
+  handleSaveFair() {
+    const {
+      relay,
+      fair: {
+        profile: { __id: fairProfile, id: fairID, is_followed: isFairFollowed },
```
Comment (l2succes) [inline src/lib/Scenes/Fair/Components/FairHeader/index.tsx]: `fairProfileID` would make more sense here

**Pull request merged**